### PR TITLE
XRDDEV-358 OpMon REST support

### PR DESCRIPTION
**Pull request by carohauta**

### CODE REVIEW

File: src/proxy/src/main/java/ee/ria/xroad/proxy/serverproxy/ServerRestMessageProcessor.java [hunk 1]
```diff
@@ -232,6 +234,19 @@ private void updateOpMonitoringDataByRequest() {
...
+
+        if (requestMessage.getRest() != null) {
+            // Set request data into the opmon object
+            opMonitoringData.setServiceId(requestServiceId);
+            opMonitoringData.setRequestAttachmentCount(decoder.getAttachmentCount());
+            opMonitoringData.setRequestRestSize(requestMessage.getRest().getMessageBytes().length);
+
+            if (decoder.getAttachmentCount() > 0) {
+                opMonitoringData.setRequestMimeSize(requestMessage.getRest().getMessageBytes().length
+                        + decoder.getAttachmentsByteCount());
```
Comment (jhyoty) [inline src/proxy/src/main/java/ee/ria/xroad/proxy/serverproxy/ServerRestMessageProcessor.java]: The decoder does not update the attachment byte count for rest bodies, so the size is wrong.

File: src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageProcessor.java [hunk 2]
```diff
@@ -152,8 +154,8 @@ private void updateOpMonitoringServiceSecurityServerAddress(URI addresses[], Htt
 
     private void updateOpMonitoringDataByResponse(ProxyMessageDecoder decoder) {
         if (response.getRestResponse() != null) {
-            opMonitoringData.setResponseSoapSize(0);
-            opMonitoringData.setResponseAttachmentCount(0);
+            opMonitoringData.setResponseRestSize(response.getRestResponse().getMessageBytes().length);
+            opMonitoringData.setResponseAttachmentCount(decoder.getAttachmentCount());
         }
```
Comment (jhyoty) [inline src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageProcessor.java]: Updating total size (mime size) missing?

File: src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageProcessor.java [hunk 3]
```diff
@@ -367,4 +369,18 @@ public boolean isStreaming() {
...
+     * the size of the message.
+     */
+    private void updateOpMonitoringDataByRestRequest() {
+        if (opMonitoringData != null && restRequest != null) {
+            opMonitoringData.setClientId(senderId);
+            opMonitoringData.setServiceId(requestServiceId);
+            opMonitoringData.setMessageId(restRequest.findHeaderValueByName(MimeUtils.HEADER_QUERY_ID));
+            opMonitoringData.setMessageUserId(restRequest.findHeaderValueByName(MimeUtils.HEADER_USER_ID));
+            opMonitoringData.setMessageIssue(restRequest.findHeaderValueByName(MimeUtils.HEADER_ISSUE));
+        }
```
Comment (jhyoty) [inline src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageProcessor.java]: request size missing?

File: src/proxy/src/main/java/ee/ria/xroad/proxy/serverproxy/ServerRestMessageProcessor.java (hunk 1 above)
Comment (carohauta) [reply src/proxy/src/main/java/ee/ria/xroad/proxy/serverproxy/ServerRestMessageProcessor.java]: Decoder now updates the attachment byte count

**Pull request merged**